File: Data Structures & Algorithms/validate-parentheses/submission-3.py
class Solution:
    def isValid(self, s: str) -> bool:
        l=deque()
        for i in s:
            if len(l)!=0:
                if i=='}' and '{'==l[-1]:
                    l.pop()
                elif i==']' and '['==l[-1]:
                    l.pop()
                elif i==')' and '('==l[-1]:
                    l.pop()
                else:
                    l.append(i)
            else:
                l.append(i)
        if len(l)==0:
            return True
        else:
            return False

File: Data Structures & Algorithms/asteroid-collision/submission-3.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& asteroids) {
        stack<int>st;
        for(int i:asteroids){
            if(st.empty()) st.push(i);
            else if(i<0){
                if(st.top()<0)st.push(i);
                else{
                    while(!st.empty()&&-st.top()>i&&st.top()>0)st.pop();
                    if(!st.empty()&&st.top()==(-i))st.pop();
                    else if(st.empty() || st.top()<0)st.push(i);
                }
            }
            else{
                st.push(i);
            }
        }
        vector<int>vec;
        while(!st.empty()){
            vec.push_back(st.top());
            st.pop();
        }
        reverse(vec.begin(),vec.end());
        return vec;
    }
};